frida-gum/src/interceptor.rs

```diff
@@ -21,19 +21,36 @@ mod invocation_listener;
 pub use invocation_listener::*;
 
 /// Function hooking engine interface.
-pub struct Interceptor<'a> {
+pub struct Interceptor {
     interceptor: *mut gum_sys::GumInterceptor,
-    phantom: PhantomData<&'a gum_sys::GumInterceptor>,
+    _gum: Gum,
 }
 
-impl<'a> Interceptor<'a> {
+impl Clone for Interceptor {
+    fn clone(&self) -> Self {
+        Interceptor {
+            interceptor: unsafe {
+                frida_gum_sys::_frida_g_object_ref(self.interceptor as *mut _) as *mut _
```

Comment:
You use `_frida_g_object_ref` here, but `g_object_unref` in the `Drop` implementation below. Let's try to be consistent. In any event I think we need wrapper functions for these functions, as they will have different names on different platforms, as per @oleavr's comment.

Reply:
Done.